packages/react-native-session-replay/android/src/newarch/kotlin/com/datadog/sessionreplay/reactnative/DdSessionReplay.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.sessionreplay.reactnative
+
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactMethod
+
+/**
+ * The entry point to use Datadog's Session Replay feature.
+ */
+class DdSessionReplay(
+    reactContext: ReactApplicationContext
+) : NativeDdSessionReplaySpec(reactContext) {
+
+    private val implementation = DdSessionReplayImplementation()
```

Comment:
why do we need a dedicated implementation class? what is its purpose?

UPD: is it to have a single class which can be used from both old arch and new arch plugins?

Reply:
Yes exactly, this is not great but is the current standard when supporting both old and new architecture in RN libs.